### PR TITLE
Docs: Plugin E2E - Config page article

### PR DESCRIPTION
**Pull request by sunker**
**What this PR does / why we need it**:

This is the first article in the `Test a data source plugin` guide. Eventually I'd like to have the following articles:
```
├── test-a-data-source-plugins.md 
│   ├── configuration-editor.md
│   ├── query-editor.md
│   ├── custom-variable-editor.md
│   ├── template-variable-interpolation.md
│   ├── annotation-editor.md
│   ├── alerting.md
```
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/configuration-editor.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+}) => {
+  const ds = readProvisionedDataSource<JsonData, SecureJsonData>({ fileName: 'datasources.yaml' });
+  const configPage = await createDataSourceConfigPage({ type: ds.type });
+  await page.getByLabel('Path').fill(ds.jsonData.path);
+  await page.getByLabel('API Key').fill(ds.secureJsonData.apiKey);
+  await expect(configPage.saveAndTest()).toBeOK();
+});
+```
+
+### When configuration is invalid
```
Comment (sympatheticmoose) [inline docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/configuration-editor.md:38]: if this is backend specific then should be h4, if not, might be better after the frontend section?

Comment (sunker) [reply docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/configuration-editor.md]: It's backend specific, so changing to h4 👍

**Review by mckn — APPROVED** [review]
LGTM! Great work on this!

**Review by sympatheticmoose — COMMENTED** [review]
Some comments, but I've not been able to successfully run the tests yet and would like to give that a go prior to approval

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/configuration-editor.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  - plugins
+  - plugin
+  - testing
+  - e2e
+  - data-source
+  - configuration editor
+sidebar_position: 10
+---
+
+## Introduction
```
[the remark this answers is not in the document]
Comment (sunker) [reply docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/configuration-editor.md:17]: yep I'll remove

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/configuration-editor.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  const configPage = await createDataSourceConfigPage({ type: ds.type });
+  await page.getByLabel('Path').fill(ds.jsonData.path);
+  await expect(configPage.saveAndTest()).not.toBeOK();
+  await expect(configPage).toHaveAlert('error', { hasText: 'API key is missing' });
+});
+```
+
+### Testing the configuration in a frontend data source plugin
+
+A frontend data source plugin may call any endpoint to test whether the provided configuration is valid. You can use Playwright's [`waitForResponse`](https://playwright.dev/docs/api/class-page#page-wait-for-response) method and specify the url of the endpoint that is being called.
```
[the remark this answers is not in the document]
Comment (sunker) [reply docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/configuration-editor.md]: Made an attempt to improve this. Not sure if it's any better?

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/configuration-editor.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+```ts title="configurationEditor.spec.ts"
```
[the remark this answers is not in the document]
Comment (sunker) [reply docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/configuration-editor.md:62]: The snippets get really long, so only include the imports in the first one. Think that's ok

[the remark this answers is not in the document]
Comment (sunker) [reply docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/configuration-editor.md:63]: @sympatheticmoose the suggested way to test frontend data sources was weird, so I just merged a [PR](https://github.com/grafana/plugin-tools/pull/785) that improves this slightly. Have updated the snippets accordingly.

**Review by josmperez — APPROVED** [review]
LGTM.

**Pull request merged**